src/solver/simulation/solver.hxx

```diff
@@ -344,7 +345,10 @@ void ISimulation<Impl>::run()
         }
 
         if (parameters.useCustomScenario)
+        {
             ApplyCustomScenario(study);
+            FinalReservoirLevelPreChecks(study);
```

Comment:
To complete the [above comment](https://github.com/AntaresSimulatorTeam/Antares_Simulator/pull/1175/files?show-viewed-files=true&file-filters%5B%5D=#r1112070312) : 
Indeed, `FinalReservoirLevelPreChecks(study);` should be called here, but in **Rules::readFinalHydroLevels**, see this comment